api/src/test/java/org/openmrs/api/ProgramWorkflowServiceTest.java

```diff
@@ -479,6 +506,100 @@ public void getProgramByUuid_shouldFindObjectGivenValidUuid() {
 	public void getProgramByUuid_shouldReturnNullIfNoObjectFoundWithGivenUuid() {
 		Assert.assertNull(Context.getProgramWorkflowService().getProgramByUuid("some invalid uuid"));
 	}
+
+	/**
+	 * @see ProgramWorkflowService#getPrograms(String)
+	 */
+	@Test
+	public void getPrograms_shouldGetProgramByNameFragment() {
+		List<Program> allPrograms = Context.getProgramWorkflowService().getAllPrograms();
+		int programCount=0;
+		final String programName = "program name";
```

Comment:
I suggest you remove this logic and simply do

```java
List<Program> programs = pws.getPrograms("program name");

assertEquals(xxx, programs.size())
```

by putting in the number you expect.